oemer/ete.py

```diff
@@ -142,7 +142,13 @@ def extract(args: Namespace) -> str:
             pickle.dump(data, open(pkl_path, "wb"))
 
     # Load the original image, resize to the same size as prediction.
-    image = cv2.imread(str(img_path))
+    if not str(img_path).endswith(".gif"):
+        image = cv2.imread(str(img_path))
+    else:
+        gif_image = Image.open(str(img_path)).convert('RGB')
+        gif_image = np.array(gif_image)
+        image = gif_image[:, :, ::-1].copy()
```

Comment:
Is there any specific reason to shuffle the RGB channel?

Reply:
PIL.Image() uses RGB, OpenCV uses BGR. I shuffle the channels to ensure the GIF image has the same order of channels as the others, which may matter in the conversion to grayscale where each channel has a different weight (but I haven't checked in the code whether and where this happens).

Reply:
I see. Thanks for the explanation.